Helix passes `null` in `previousResultId` for `textDocument/diagnostic`

### Summary


Originally reported in: https://github.com/microsoft/typescript-go/issues/3336

The spec says:

```ts
export interface DocumentDiagnosticParams extends WorkDoneProgressParams,
	PartialResultParams {
	textDocument: TextDocumentIdentifier;
	identifier?: string;
	previousResultId?: string;
}
```

`null` is not allowed for `previousResultId`; editor should not be passing it.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#documentDiagnosticParams

### Reproduction Steps

1. open up a project with tsgo using `@typescript/native-preview@7.0.0-dev.20260403.1`
2. see that there are no diagnostics
3. look in the helix log and see the error

### Helix log

`2026-04-03T19:10:31.323 helix_lsp::transport [INFO] tsgo -> {"jsonrpc":"2.0","method":"textDocument/diagnostic","params":{"previousResultId":null,"textDocument":{"uri":"file:///test/src/App.tsx"}},"id":4}`

`2026-04-03T19:10:31.323 helix_lsp::transport [ERROR] tsgo <- InvalidParams: InvalidParams: json: cannot unmarshal JSON object into Go lsproto.Message: InvalidParams: failed to unmarshal *lsproto.DocumentDiagnosticParams: json: cannot unmarshal into Go lsproto.DocumentDiagnosticParams within "/previousResultId": null value is not allowed for field "previousResultId"`


### Platform

Linux

### Terminal Emulator

foot version: 1.26.1 +pgo +ime +graphemes +toplevel-tag +blur -assertions

### Installation Method

flake

### Helix Version

helix 25.07.1 (daac117b)

--- a/helix-lsp-types/src/document_diagnostic.rs
+++ b/helix-lsp-types/src/document_diagnostic.rs
@@ -108,6 +108,7 @@ pub struct DocumentDiagnosticParams {
     pub identifier: Option<Arc<str>>,
 
     /// The result ID of a previous response if provided.
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub previous_result_id: Option<String>,
 
     #[serde(flatten)]

